### PR TITLE
Prerequisite changes of feature/compression

### PR DESCRIPTION
**Pull request by jesec**
Compression change (#1200) is giant. It would be easier for us to split, review and merge the related changes first.

### CODE REVIEW

File: prelude/bootstrap.js [hunk 1]
```diff
@@ -1713,9 +1720,9 @@ function payloadFileSync(pointer) {
     // /////////////////////////////////////////////////////////////
 
     Object.defineProperty(require('fs').exists, custom, {
-      value(path) {
+      value(path_) {
         return new Promise((resolve) => {
-          require('fs').exists(path, (exists) => {
+          require('fs').exists(path_, (exists) => {
```
Comment (erossignon) [inline prelude/bootstrap.js:1725]: => fs.exists

File: prelude/bootstrap.js [hunk 2]
```diff
@@ -1713,9 +1720,9 @@ function payloadFileSync(pointer) {
     // /////////////////////////////////////////////////////////////
 
     Object.defineProperty(require('fs').exists, custom, {
```
Comment (erossignon) [inline prelude/bootstrap.js:1722]: require('fs') => fs

File: prelude/bootstrap.js [hunk 3]
```diff
@@ -1642,12 +1652,9 @@ function payloadFileSync(pointer) {
       args[0] === EXECPATH
     ) {
       args[0] = EXECPATH;
-      if (NODE_VERSION_MAJOR === 0) {
```
Comment (erossignon) [inline prelude/bootstrap.js:1645]: Shall this go to a different commit ?

File: prelude/bootstrap.js [hunk 4]
```diff
@@ -91,23 +93,22 @@ EXECSTAT.birthtimeMs = EXECSTAT.birthtime.getTime();
 // MOUNTPOINTS /////////////////////////////////////////////////////
 // /////////////////////////////////////////////////////////////////
 
-var mountpoints = [];
+const mountpoints = [];
 
 function isRegExp(val) {
   return require('util').types.isRegExp(val);
```
Comment (erossignon) [inline prelude/bootstrap.js:99]: const {isRegExp} = require('util').types;

File: prelude/bootstrap.js [hunk 5]
```diff
@@ -1435,29 +1462,12 @@ function payloadFileSync(pointer) {
...
-      };
-      rqfn.main = process.mainModule;
-      rqfn.extensions = Module._extensions;
-      rqfn.cache = Module._cache;
-      return rqfn;
-    };
-  } else if (NODE_VERSION_MAJOR <= 9) {
+  if (NODE_VERSION_MAJOR <= 9) {
     im = require('internal/module');
-    if (NODE_VERSION_MAJOR <= 7) {
```
Comment (erossignon) [inline prelude/bootstrap.js:1456]: Shall we move to a differnt commit ? this is changing logic

**Review by erossignon — CHANGES_REQUESTED** [review]

**Review by erossignon — CHANGES_REQUESTED** [review]
I see a few 'var' left in boostrap.js that we could get rid off while we are here.

May be we could replace `path_`  with either `filename`  `pathname` or `foldername` as appropriate.

**Comment by jesec** [comment]
> I see a few 'var' left in boostrap.js that we could get rid off while we are here.
> 
> May be we could replace `path_` with either `filename` `pathname` or `foldername` as appropriate.

Done. 

I think `path` is best fit here.

**Review by robertsLando — APPROVED** [review]

**Review by erossignon — APPROVED** [review]

**Pull request merged**